### ISSUE TITLE
Feature request: per-curve user attributes from Maya exporting bundled curves (riCurves)

### ISSUE THREAD

**Issue by mike1158** [alembic/alembic#164]
Exporting curves in a single group in Maya with riCurves=true currently ignores any per-curve user (dynamic) attributes.  It'd be very useful if they were preserved.

**Comment by lamiller0** [comment]
Is it feasible to collect those attrs and set it on the MFnTransform which represents the curve group?

**Comment by mike1158** [comment]
That doesn't strike me as the "right" way to do it--the data are really properties of the curves themselves, not the transform.  That said, it would get the data into the file!  Thanks for the suggestion.

**Comment by lamiller0** [comment]
Agreed, it's not the ideal solution, but a lot of extra code would need to be introduced to gather all these extra properties.  Also, imagine the headache if some of the properties didn't exist on some of the curves, happened to be a different scope on some of them, or even were different data types.

A better solution would be if Maya had a single DAG node that represented multiple curves. (does it?)

**Comment by mike1158** [comment]
Yeah, I was thinking Alembic should be pretty strict about curves with mismatching attribute schemas--just dropping the mismatching curve or attribute, or failing the whole export.  I don't think Maya has a generic "bundle of curves" node, but I sure wish it did...

Anyway, this isn't a high-priority request.  I mostly threw it out to see if there was anyone else interested or maybe already had an internal implementation they might contribute.

**Comment by robotmitchum** [comment]
I encountered this issue today the hard way when I was not really expecting it...
When I export a bundle of curve (multiple curve shapes parented under a single transform) on a single frame my custom attributes on shapes are kept but as soon as I export an animation, they go away.
I would also be very interested to have this feature.
